pkg/volume/azure_dd/controllerCommon.go

```diff
@@ -0,0 +1,520 @@
+
+	vhdBlobUriTemplate             = "https://%s.blob.core.windows.net/%s/%s"
+	storageAccountUriTemplate      = "https://%s.blob.core.windows.net/%s?%s"
+	storageAccountNameTemplate     = "pvc%s"
+	storageAccountEndPointTemplate = "%ssubscriptions/%s/resourcegroups/%s/providers/microsoft.storage/storageaccounts/%s?api-version=2016-01-01"
+
+	// for limits check https://docs.microsoft.com/en-us/azure/azure-subscription-service-limits#storage-limits
+	maxStorageAccounts                     = 100 // max # is 200 (250 with special request). this allows 100 for everything else including stand alone disks
+	maxDisksPerStorageAccounts             = 60
+	storageAccountUtilizationBeforeGrowing = 0.5
+	storageAccountsCountInit               = 2 // When the plug-in is init-ed, 2 storage accounts will be created to allow fast pvc create/attach/mount
+)
+
+var defaultBackOff = kwait.Backoff{
+	Steps:    20,
+	Duration: 2 * time.Second,
+	Factor:   1.5,
+	Jitter:   0.0,
+}
+
+var time1970 time.Time = time.Date(1970, 1, 1, 0, 0, 0, 0, time.UTC)
+
+type controllerCommon struct {
+	tenantId            string
+	subscriptionId      string
+	location            string
+	resourceGroup       string
+	clientId            string
+	clientSecret        string
+	managementEndpoint  string
+	tokenEndPoint       string
+	aadResourceEndPoint string
+	aadToken            string
+	expires_on          time.Time
+}
+
+func (c *controllerCommon) isManagedArmVm(storageProfile map[string]interface{}) bool {
+	osDisk := storageProfile["osDisk"].(map[string]interface{})
+
+	_, ok := osDisk["managedDisk"]
+	if ok {
+		return true
+	}
+
+	return false
+
+}
+func (c *controllerCommon) getAttachedDisks(nodeName string) ([]string, error) {
+	var disks []string
+	var vmData interface{}
+	vm, err := c.getArmVm(nodeName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err := json.Unmarshal(vm, &vmData); err != nil {
+		return disks, err
+	}
+
+	fragment := vmData.(map[string]interface{})
+
+	props := fragment["properties"].(map[string]interface{})
+	storageProfile := props["storageProfile"].(map[string]interface{})
+	dataDisks, _ := storageProfile["dataDisks"].([]interface{})
+
+	// we silently ignore, if VM does not have the disk attached
+	for _, v := range dataDisks {
+		d := v.(map[string]interface{})
+		if _, ok := d["vhd"]; ok {
+			// this is a blob disk
+			vhdInfo := d["vhd"].(map[string]interface{})
+			vhdUri := vhdInfo["uri"].(string)
+			disks = append(disks, vhdUri)
+		} else {
+			// this is managed disk
+			managedDiskInfo := d["managedDisk"].(map[string]interface{})
+			managedDiskId := managedDiskInfo["id"].(string)
+			disks = append(disks, managedDiskId)
+		}
+	}
+	return disks, nil
+}
+
+// if disk attached returns bool + lun attached to
+func (c *controllerCommon) isDiskAttached(hashedDiskUri, nodeName string, isManaged bool) (bool, int, error) { //attached, lun (string), error
```

Comment:
Rather than a comment, just do:

`(attached bool, lun int, err error)`

as the return